metacat-thrift/src/main/java/com/netflix/metacat/thrift/CatalogThriftHiveMetastore.java

```diff
@@ -976,7 +1006,29 @@ public GetOpenTxnsInfoResponse get_open_txns_info() throws TException {
     public List<PartitionSpec> get_part_specs_by_filter(final String dbName, final String tblName,
                                                         final String filter, final int maxParts)
         throws TException {
-        throw unimplemented("get_part_specs_by_filter", new Object[]{dbName, tblName, filter, maxParts});
+        //TODO: Handle the use case of grouping
+        return requestWrapper("get_partitions_pspec", new Object[]{dbName, tblName, filter, maxParts}, () -> {
+            final String databaseName = normalizeIdentifier(dbName);
+            final String tableName = normalizeIdentifier(tblName);
+            final TableDto tableDto = v1.getTable(catalogName, databaseName, tableName, true, false, false);
+
+            final List<PartitionDto> metacatPartitions = partV1.getPartitions(catalogName, dbName, tblName,
+                filter, null,
+                null, null, maxParts, false);
+            final List<Partition> partitions = Lists.newArrayListWithCapacity(metacatPartitions.size());
+            for (PartitionDto partition : metacatPartitions) {
+                partitions.add(hiveConverters.metacatToHivePartition(partition, tableDto));
+            }
+
+            final PartitionSpec pSpec = new PartitionSpec();
+            pSpec.setPartitionList(new PartitionListComposingSpec(partitions));
+            pSpec.setDbName(dbName);
+            pSpec.setTableName(tblName);
+            if (tableDto != null && tableDto.getSerde() != null) {
+                pSpec.setRootPath(tableDto.getSerde().getUri());
+            }
```

Comment:
I am not sure if this is correct. The partition uri can be different from the table serde uri.  If just for information purpose it's fine, I guess.

Reply:
I have taken this logic from Hive Metastore code.